From ba564329cdc2a66f461d8aa2992193bbd606fe62 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:38:33 -0400
Subject: [ethernet] i40e: clear pxe after adminq is rebuilt

Message-id: <1409658042-22689-153-git-send-email-sassmann@redhat.com>
Patchwork-id: 90345
O-Subject: [RHEL7.1 PATCH 152/281] i40e: clear pxe after adminq is rebuilt
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shannon Nelson <shannon.nelson@intel.com>

Be sure to clear PXE mode bit on each reset after AdminQ has been rebuilt.

Change-ID: I992d8c79594f8ca0660c50844ace675ecb9c9bf2
Signed-off-by: Shannon Nelson <shannon.nelson@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit e78ac4bff2f9791f479ebfa2c1a496087b321e99)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index a3c5ce4..c68d821 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -5520,6 +5520,7 @@ static void i40e_reset_and_rebuild(struct i40e_pf *pf, bool reinit)
   i40e_verify_eeprom(pf);
  }
 
+ i40e_clear_pxe_mode(hw);
  ret = i40e_get_capabilities(pf);
  if (ret) {
   dev_info(&pf->pdev->dev, "i40e_get_capabilities failed, %d\n",
-- 
1.7.1